sycl/doc/design/Fuzzing.md

```diff
@@ -0,0 +1,104 @@
+# Fuzzing plan for intel/llvm
+
+The main product which is being developed at intel/llvm repo is a SYCL
+implementation. At high-level, it consists of two compiler and runtime and
+therefore this document will be divided into two major sections covering those
+components. Those components are essentially the only entry points through which
+a user can interact with our product.
```

Comment:
It would make sense to add a short summary of what fuzzing is and why we want to do it.